### DISCUSSION TITLE
Constraints of Command selection screen

### DISCUSSION THREAD

**Discussion by DaiSugi01** [DaiSugi01/Queens-game#54] [Ideas]
I'm working on making the screen and I realized that the screen title doesn't show correctly on SE 2nd gene when both leading and trailing are set 32 like below pic.

Since I can fix it by changing the side margins into 26, I'm thinking to make computed properties like the top and bottom constraints in Constant.Common if screen width is less than 400 so that we can handle with SE and 12 mini.

What would you guys think?

[Device info](https://qiita.com/tomohisaota/items/f8857d01f328e34fb551)

### preview(leading: 32, trailing: -32)
![SE](https://user-images.githubusercontent.com/37020406/118046765-a4d19c00-b32e-11eb-8b74-80930e8581e4.png)

### preview(leading: 26, trailing: -26)
![se (1)](https://user-images.githubusercontent.com/37020406/118049885-422ecf00-b333-11eb-95f0-106470fb9e30.png)

**Comment by cookie777** [comment]
Thank's for finding.
That's good idea.

I'm also thinking changing the font size depending on screen size.
Change `64` to some variable which varies depends on screen.
```swift
  static var h12 UIFont = {
    let f = UIFont.systemFont(ofSize: 64, weight: .black)
    return f
  }()
```

This is because it seams font size is too big and space is a little bit narrow in SE device.

**Reply by DaiSugi01** [reply]
Can you try to change it first? 
If it still doesn't display the whole title, I'd change the side constraints.

I think the title on the CommandSelection screen is the longest, so we can check if it's displayed correctly.
("How do make your command?")

**Comment by cookie777** [comment]
How about doing these two things

1. change the number of lines to `0`
2. change font size depends on screen

By this, it will shrink and looks nice in small screeen.
Besides it will also display all text.



```swift
  let screenTitle: H2Label = {
    let lb = H2Label(text: "How do you make your command?")
    lb.translatesAutoresizingMaskIntoConstraints = false
    lb.lineBreakMode = .byWordWrapping
    lb.numberOfLines = 0
    lb.setContentHuggingPriority(.required, for: .vertical)
    return lb
  }()
```


```swift
  static var h2: UIFont = {
    let fontSize = UIScreen.main.bounds.height <= 700 ? 32 : 40
    let f = UIFont.systemFont(ofSize: CGFloat(fontSize), weight: .black)
    return f
  }()
```
![image](https://user-images.githubusercontent.com/41662956/118054319-e9633480-b33a-11eb-9c0f-3958ffaf7674.png)
![image](https://user-images.githubusercontent.com/41662956/118054441-1d3e5a00-b33b-11eb-800b-b46f9e522612.png)

**Reply by DaiSugi01** [reply]
That's cool. 
I tried it on my local repository and it works well too. (I don't change constraints for both leading and trailing which is 32)
Can you push it to the master branch?
## SE
![SE1](https://user-images.githubusercontent.com/37020406/118069962-ef1c4280-b359-11eb-9526-67d224be87ec.png)
![SE2](https://user-images.githubusercontent.com/37020406/118070021-078c5d00-b35a-11eb-825b-3469766cab81.png)

## 12 pro
![12pro2](https://user-images.githubusercontent.com/37020406/118070150-399dbf00-b35a-11eb-952a-c0fd0f63f9d6.png)
![12pro1](https://user-images.githubusercontent.com/37020406/118070103-2ab70c80-b35a-11eb-83b0-ff062ff98a7e.png)

**Comment by cookie777** [comment]
I added responsive font size in custom font.
If you want to add more section (eg, 700...800), you can just add another `case`.


```swift
  static var h2: UIFont = {
    let f = UIFont.systemFont(ofSize: h2FontSize, weight: .black)
    return f
  }()

  static private var h2FontSize: CGFloat {
    switch UIScreen.main.bounds.height {
      case 0...700:
        return 32
      default:
        return 40
    }
  }
```
